src/game/logic/system/gamelogic.cpp

```diff
@@ -390,3 +609,2356 @@ GhostObjectManager *GameLogic::Create_Ghost_Object_Manager()
+        m_sleepingUpdateModules[0]->Set_Index_In_Logic(-1);
+
+        if (size <= 1) {
+            m_sleepingUpdateModules.pop_back();
+        } else {
+            m_sleepingUpdateModules[0] = m_sleepingUpdateModules[size - 1];
+            m_sleepingUpdateModules[0]->Set_Index_In_Logic(0);
+            m_sleepingUpdateModules.pop_back();
+            Rebalance_Child_Sleepy_Update(0);
+        }
+    } else {
+        captainslog_dbgassert(false, "should not happen");
+    }
+}
+
+unsigned int GameLogic::Get_CRC(int i, Utf8String s)
+{
+    if (i == 1) {
+        Set_FP_Mode();
+        static bool m_inCRCGen;
+        bool b = Get_In_Game_Logic_Update();
+        LatchRestore<bool> latch(&m_inCRCGen, &b);
+        Utf8String str;
+        XferCRC *xfer;
+
+        if (s.Is_Not_Empty()) {
+            xfer = new XferDeepCRC();
+            xfer->Open(s.Str());
+        } else {
+            Utf8String name;
+#ifdef GAME_DEBUG_STRUCTS
+            // TODO deep CRC stuff
+#endif
+            xfer = new XferCRC();
+            name = "lightCRC";
+            xfer->Open(name.Str());
+        }
+
+        captainslog_dbgassert(this == g_theGameLogic, "Not in GameLogic");
+
+#ifdef GAME_DEBUG_STRUCTS
+        // TODO unknown CRC stuff
+#endif
+        str = "MARKER:Objects";
+        xfer->xferAsciiString(&str);
+
+        for (Object *obj = m_objList; obj != nullptr; obj = obj->Get_Next_Object()) {
+            xfer->xferSnapshot(obj);
+        }
+
+        unsigned int logic_crc = Get_Logic_Random_Seed_CRC();
+
+#ifdef GAME_DEBUG_STRUCTS
+        // TODO unknown CRC stuff
+#endif
+
+        if (xfer->Get_Mode() == XFER_CRC) {
+            xfer->xferUnsignedInt(&logic_crc);
+        }
+
+        str = "MARKER:ThePartitionManager";
+        xfer->xferAsciiString(&str);
+        xfer->xferSnapshot(g_thePartitionManager);
+
+#ifdef GAME_DEBUG_STRUCTS
+        // TODO unknown CRC stuff
+#endif
+
+        str = "MARKER:ThePlayerList";
+        xfer->xferAsciiString(&str);
+        xfer->xferSnapshot(g_thePlayerList);
+
+#ifdef GAME_DEBUG_STRUCTS
+        // TODO unknown CRC stuff
+#endif
+        str = "MARKER:TheAI";
+        xfer->xferAsciiString(&str);
+        xfer->xferSnapshot(g_theAI);
+
+#ifdef GAME_DEBUG_STRUCTS
+        // TODO unknown CRC stuff
+#endif
+
+        if (xfer->Get_Mode() == XFER_SAVE) {
+            str = "MARKER:GameSave";
+            xfer->xferAsciiString(&str);
+            g_theGameState->Friend_Xfer_Save_Data_For_CRC(xfer, SNAPSHOT_TYPE_UNK2);
+        }
+
+        xfer->Close();
+        unsigned int crc = xfer->Get_CRC();
+        delete xfer;
+        xfer = nullptr;
+
+#ifdef GAME_DEBUG_STRUCTS
+        // TODO unknown CRC stuff
+#endif
+
+        return crc;
+    } else {
+        return m_crc;
+    }
+}
+
+void GameLogic::Set_Game_Paused(bool b1, bool b2)
+{
+    if (b1 != m_gamePaused) {
+        m_gamePaused = b1;
+        int sound = b2 + (AUDIOAFFECT_SPEECH | AUDIOAFFECT_3DSOUND | AUDIOAFFECT_SOUND);

```

Comment:
Looks like compiler optimization.

Corrected:

```c++
int audio_affect = (pause_music ? AUDIOAFFECT_MUSIC : 0) | AUDIOAFFECT_SPEECH | AUDIOAFFECT_3DSOUND | AUDIOAFFECT_SOUND;
```

because otherwise this breaks if `AUDIOAFFECT_MUSIC` is no longer 1.

Reply:
fixed.